src/components/button-toggle/button-toggle-group/button-toggle-group.test.tsx

```diff
@@ -0,0 +1,424 @@
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      label="Group Label"
+      inputHint="Group Hint Text"
+      onChange={() => {}}
+      disabled
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("button", { name: "Foo" })).toBeDisabled();
+  expect(screen.getByRole("button", { name: "Bar" })).toBeDisabled();
+  expect(screen.getByRole("group")).toHaveStyle({
+    cursor: "not-allowed",
+    boxShadow: "inset 0px 0px 0px 1px var(--colorsActionDisabled600)",
+  });
+  expect(screen.getByText("Group Hint Text")).toHaveStyle({
+    color: "var(--colorsUtilityYin030)",
+  });
+});
+
+test("should render with expected styles when fullWidth prop is set", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      label="Group Label"
+      onChange={() => {}}
+      fullWidth
+    >
+      <ButtonToggle value="foo" data-role="foo">
+        Foo
+      </ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("group")).toHaveStyle({ width: "100%" });
+  expect(screen.getByTestId("foo")).toHaveStyle({ flex: "auto" });
+});
+
+test("should render with expected styles when inputWidth prop is set", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      label="Group Label"
+      onChange={() => {}}
+      inputWidth={50}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("group")).toHaveStyle({ width: "50%" });
+});
+
+test("should render with expected styles when labelInline prop is set", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      label="Group Label"
+      onChange={() => {}}
+      labelInline
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("group")).toHaveStyle({ flexWrap: "nowrap" });
+});
+
+test("should not render labelHelp or filedHelp when validationRedesignOptIn is true", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <ButtonToggleGroup
+        id="button-toggle-group-id"
+        label="Group Label"
+        labelHelp="Label Help"
+        fieldHelp="Field Help"
+        onChange={() => {}}
+      >
+        <ButtonToggle value="foo">Foo</ButtonToggle>
+        <ButtonToggle value="bar">Bar</ButtonToggle>
+      </ButtonToggleGroup>
+    </CarbonProvider>
+  );
+
+  expect(screen.queryByText("Label Help")).not.toBeInTheDocument();
+  expect(screen.queryByText("Field Help")).not.toBeInTheDocument();
+});
+
+test("should only allow the first button to be tabbable when none are selected", async () => {
```

Comment:
**suggestion**: for these next two tabbable tests, how about performing the tab actions and verifying that the focus is set correctly, instead of checking the tabIndex? This approach should achieve the same result, but will more closely mimic user behavior.

```tsx
test("should only allow the first button to be tabbable when none are selected", async () => {
  // render and setup...

  await user.tab();

  expect(screen.getByRole("button", { name: "Foo" })).toHaveFocus();

  await user.tab();

  expect(screen.getByRole("button", { name: "Bar" })).not.toHaveFocus();
});
```